src/controllers/wallet/atomic-swap/tx-proposal.controller.js

```diff
@@ -0,0 +1,292 @@
+    && sendTokens.length === 0
+    && receiveTokens.length === 0
+  ) {
+    res.status(400).json({ success: false, error: ['Should have at least one operation'] });
+    return;
+  }
+
+  const proposal = partialTx
+    ? PartialTxProposal.fromPartialTx(partialTx, network) : new PartialTxProposal(network);
+
+  for (const input of inputs) {
+    let error = null;
+    txApi.getTransaction(input.txId, data => {
+      if (!data.success) {
+        error = `Utxo for transaction ${input.txId} and input ${input.index} not found`;
+        return;
+      }
+
+      const txout = data.tx.outputs[input.index];
+      proposal.addInput(
+        req.wallet,
+        txout.txId,
+        txout.index,
+        txout.value,
+        txout.decoded.address,
+        {
+          markAsSelected,
+          token_data: txout.token_data,
+          token: txout.token,
+        },
+      );
+    });
+
+    if (error) {
+      res.status(400).json({ success: false, errors: [error] });
+      return;
+    }
+  }
+
+  for (const send of sendTokens) {
+    const token = send.token || HATHOR_TOKEN_CONFIG.uid;
+    proposal.addSend(req.wallet, token, send.value, { changeAddress, markAsSelected });
+  }
+
+  for (const output of outputs) {
+    const token = output.token || HATHOR_TOKEN_CONFIG.uid;
+    proposal.addOutput(token, output.value, output.address);
+  }
+
+  for (const receive of receiveTokens) {
+    const token = receive.token || HATHOR_TOKEN_CONFIG.uid;
+    const timelock = receive.timelock || null;
+    const address = receive.address || null;
+    proposal.addReceive(
+      req.wallet,
+      token,
+      receive.value,
+      { timelock, address }
+    );
+  }
+
+  res.send({
+    success: true,
+    data: proposal.partialTx.serialize(),
+    isComplete: proposal.partialTx.isComplete(),
+  });
+}
+
+async function getMySignatures(req, res) {
+  const validationResult = parametersValidation(req);
+  if (!validationResult.success) {
+    res.status(400).json(validationResult);
+    return;
+  }
+
+  const network = req.wallet.getNetworkObject();
+  const partialTx = req.body.partial_tx;
+  const proposal = PartialTxProposal.fromPartialTx(partialTx, network);
+
+  try {
+    storage.setStore(req.wallet.store);
+    await proposal.signData('123');
+    res.send({
+      success: true,
+      signatures: proposal.signatures.serialize(),
+      isComplete: proposal.signatures.isComplete(),
+    });
+  } catch (err) {
+    res.send({ success: false, error: err.message });
+  }
+}
+
+async function signTx(req, res) {
+  const validationResult = parametersValidation(req);
+  if (!validationResult.success) {
+    res.status(400).json(validationResult);
+    return;
+  }
+
+  const network = req.wallet.getNetworkObject();
+
+  const partialTx = req.body.partial_tx;
+  const signatures = req.body.signatures || [];
+
+  const proposal = PartialTxProposal.fromPartialTx(partialTx, network);
+
+  try {
+    await proposal.signData('123');
```

Comment:
Because this is the `sign-and-push` API, we can change it to just `push` and let all signatures come from `get-my-signatures`.
I think it would be a good option as well.

Reply:
It's not only a `push` API, you should receive all signatures, in my opinion, assemble the tx, mine and push, so it's a bit more than just push but I think we should change this

Reply:
Changed the API to not sign with the loaded wallet and only assemble with the data it receives (and replicated this on the sign api).